File: app/src/main/java/fr/nextgear/mesentretiensmoto/core/views/MaintenanceCellView.java
package fr.nextgear.mesentretiensmoto.core.views;

import android.content.Context;
import android.icu.text.DateFormat;
import android.icu.text.SimpleDateFormat;
import android.view.ViewGroup;
import android.widget.TextView;

import com.afollestad.materialdialogs.MaterialDialog;
import com.orhanobut.logger.Logger;

import butterknife.BindView;
import butterknife.ButterKnife;
import fr.nextgear.mesentretiensmoto.R;
import fr.nextgear.mesentretiensmoto.core.App;
import fr.nextgear.mesentretiensmoto.core.bus.MainThreadBus;
import fr.nextgear.mesentretiensmoto.core.database.MaintenanceDBManager;
import fr.nextgear.mesentretiensmoto.core.events.EventMarkMaintenanceDone;
import fr.nextgear.mesentretiensmoto.core.model.Maintenance;
import io.nlopez.smartadapters.views.BindableLinearLayout;

public class MaintenanceCellView extends BindableLinearLayout<Maintenance> {

    //region Fields
    private static final String FORMAT = "%s H";
    private static final String DATE_FORMAT = "dd/MM/yyyy";
    private final Context mContext;

    @BindView(R.id.manage_maintenance_cell_TextView_nameMaintenance)
    TextView mTextViewNameMaintenance;
    @BindView(R.id.manage_maintenance_cell_GroupView)
    ViewGroup mLayout;
    @BindView(R.id.manage_maintenance_cell_TextView_nbHoursMaintenance)
    TextView mTextViewNbHoursMaintenance;

    @BindView(R.id.manage_maintenance_cell_TextView_dateMaintenance)
    TextView mTextViewDateMaintenance;
    //endregion

    //region Constructor
    public MaintenanceCellView(Context poContext) {
        super(poContext);
        mContext = poContext;
    }
    //endregion

    //region Lifecycle events
    @Override
    public int getOrientation() {
        return VERTICAL;
    }

    @Override
    public int getLayoutId() {
        return R.layout.manage_maintenance_cell;
    }

    @Override
    public void bind(Maintenance poMaintenance) {
        mTextViewNameMaintenance.setText(poMaintenance.nameMaintenance);
        if(poMaintenance.isDone){
            mTextViewNbHoursMaintenance.setVisibility(VISIBLE);
            mTextViewNbHoursMaintenance.setText(String.format(FORMAT,poMaintenance.nbHoursMaintenance));
        }
        mTextViewDateMaintenance.setText(android.text.format.DateFormat.format(DATE_FORMAT, poMaintenance.dateMaintenance));
        if(!poMaintenance.isDone) {
            mLayout.setOnClickListener(view -> new MaterialDialog.Builder(mContext)
                    .content(R.string.ask_maintenance_done)
                    .positiveText(R.string.yes)
                    .negativeText(R.string.no)
                    .onPositive((dialog, which) ->{
                        App.getInstance().getMainThreadBus().post(new EventMarkMaintenanceDone(poMaintenance));
                    })
                    .onNegative((dialog, which) -> Logger.e("remain not done "))
                    .build()
                    .show());
        }
    }

    @Override
    public void onViewInflated() {
        super.onViewInflated();
        ButterKnife.bind(this);
        setLayoutParams(new LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT));
    }
    //endregion
}

File: app/src/main/java/fr/nextgear/mesentretiensmoto/core/views/MaintenanceCellView.kt
package fr.nextgear.mesentretiensmoto.core.views

import android.text.format.DateFormat
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.material.Card
import androidx.compose.material.MaterialTheme
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import fr.nextgear.mesentretiensmoto.R
import fr.nextgear.mesentretiensmoto.core.model.Bike
import fr.nextgear.mesentretiensmoto.core.model.Maintenance
import java.util.*

@Composable
fun MaintenanceCellView(maintenance: Maintenance) {
    val context = LocalContext.current
    val countingMethod = maintenance.bike?.countingMethod
    Column(
        verticalArrangement = Arrangement.Center,
        modifier = Modifier.padding(8.dp)
    ) {
        maintenance.nameMaintenance?.let {
            Text(
                text = it,
                style = MaterialTheme.typography.h6
            )
        }
        if (maintenance.isDone) {
            Text(
                text = DateFormat.format("dd/MM/yyyy", Date(maintenance.dateMaintenance!!))
                    .toString(),
                style = MaterialTheme.typography.body2
            )
            Text(
                text = if (countingMethod == Bike.MethodCount.HOURS)
                    String.format("%s H", maintenance.nbHoursMaintenance)
                else
                    String.format("%s KM", maintenance.nbHoursMaintenance.toInt().toString()),
                style = MaterialTheme.typography.body2
            )
        } else {
            Card(
//                onClick = {
//                    MaterialDialog(context).show {
//                        title(R.string.ask_maintenance_done)
//                        positiveButton(R.string.yes) {
//                            MainThreadBus.post(EventMarkMaintenanceDone(maintenance))
//                        }
//                        negativeButton(R.string.no)
//                    }
//                },
                shape = MaterialTheme.shapes.small,
                modifier = Modifier.fillMaxWidth()
            ) {
                Text(
                    text = context.getString(R.string.title_mark_maintenance_done),
                    style = MaterialTheme.typography.body2,
                    textAlign = TextAlign.Center,
                    modifier = Modifier.padding(16.dp)
                )
            }
        }
    }
}